concepts/40 Angular Components/40 Common Features/10 Tree Shaking.md

```diff
@@ -10,4 +12,76 @@ If tree shaking is not configured in your application, import exports from speci
     <!--TypeScript-->
     import { DxButtonModule } from 'devextreme-angular/ui/button';
 
-[tags] angular
\ No newline at end of file
+[note]The tree shaking mechanism works only when you create the **production** bundle.
+
+## Bundle Size
+
+DevExtreme code consists of three main code groups:
+- Utilities Code
+    An application imports Utilities Code, when you add a DevExtreme component. Utilities Code is mandatory for DevExtreme components and the tree shaking mechanim does not remove it. 
```

Comment:
Can we make "it" more specific? For example, do you mean "this code"?

Reply:
Sure. Let's make it "this code".

Reply:
Thanks. Can you also remove the comma before "when"?